ui/packages/atlasmap/src/UI/dialogs/ParametersDialog.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useCallback, useEffect,
+} from "react";
+
+import {
+  Button,
+  Form,
+  FormGroup, FormSelect, FormSelectOption, Split, SplitItem,
+  TextInput,
+} from "@patternfly/react-core";
+
+import {
+  ConfirmationDialog,
+  IConfirmationDialogProps,
+} from "./ConfirmationDialog";
+import {PlusIcon, TimesIcon} from "@patternfly/react-icons";
+
+export interface IParameter {
+  name: string;
+  value: string;
+  options?: IParameterOption[];
+  hidden?: boolean;
+  hideable?: boolean;
+}
+
+export interface IParameterOption {
+  label: string;
+  value: string;
+}
+
+export interface IParametersDialogProps {
+  title: string;
+  parameters?: IParameter[];
+  isOpen: IConfirmationDialogProps["isOpen"];
+  onCancel: IConfirmationDialogProps["onCancel"];
+  onConfirm: (parameters: IParameter[]) => void;
+}
+export const ParametersDialog: FunctionComponent<IParametersDialogProps> = ({
+  title,
+  parameters: initialParameters = [],
+  isOpen,
+  onCancel,
+  onConfirm,
+}) => {
+  const initialAddParameter = 'Select...';
+
+  const [parameters, setParameters] = useState(initialParameters);
+
+  const [addParameter, setAddParameter] = useState(initialAddParameter);
+
+  const reset = useCallback(() => {
+    setParameters(initialParameters);
+  }, [initialParameters]);
+
+  const handleOnConfirm = useCallback(() => {
+    onConfirm(parameters);
+  }, [parameters, onConfirm]);
+
+  const handleOnCancel = useCallback(() => {
+    onCancel();
+    reset();
+  }, [onCancel, reset]);
+
+  const handleAddParameter = useCallback(() => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === addParameter) {
+        newParameter.hidden = false;
+        break;
+      }
+    }
+    setAddParameter(initialAddParameter);
+    setParameters(newParameters);
+  }, [parameters, addParameter]);
+
+  const handleRemoveParameter = useCallback((name) => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === name) {
+        newParameter.hidden = true;
+        break;
+      }
+    }
+    setParameters(newParameters);
+  }, [parameters]);
+
+  const handleOnChangeParameter = useCallback((value, event) => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === event.target.name) {
+        newParameter.value = value;
+        break;
+      }
+    }
+    setParameters(newParameters);
+  }, [parameters]);
+
+  const hiddenParameters = (): IParameter[] => {
+    return parameters.filter(parameter => parameter.hidden);
+  };
+
+  const visibleParameters = (): IParameter[] => {
+    return parameters.filter(parameter => !parameter.hidden);
+  };
```

Comment:
This block shouldn't be required anymore with the proposed change. As a general rule, you don't need to make these function, just compute the value directly. Eg:

```
const hiddenParameters = parameters.filter(parameter => parameter.hidden)
```

If it's an expensive operation you can wrap it with `useMemo` to compute it only when the dependencies change. Eg:

```
const hiddenParameters = useMemo(() => parameters.filter(parameter => parameter.hidden, [parameters]); // runs only when parameters changes
```